lib/spack/llnl/util/lock.py

```diff
@@ -241,16 +309,21 @@ def acquire_read(self, timeout=None):
         """
         timeout = timeout or self.default_timeout
 
+        locktype = 'READ LOCK'
         if self._reads == 0 and self._writes == 0:
-            self._debug(
-                'READ LOCK: {0.path}[{0._start}:{0._length}] [Acquiring]'
-                .format(self))
+            self._log_acquiring(locktype)
```

Comment:
These log statements clutter the code in `acquire_read` and `acquire_write` -- why not just put them in `_lock()`?  `_lock()` can also just look up the lock type in `lock_type`, which would eliminate the need to write out `locktype = 'READ LOCK'`, which feels redundant.

Reply:
Moved `_log_acquiring` calls to `_lock`.  Did not move the `log_acquired` since they are dependent on an updated `_read`/`_write` count (when verbose).